### PR TITLE
[DM] Patch IAM member to projectID

### PR DESCRIPTION
**Pull request by ocsig**
Project ID was forcefully overwritten to context.env['project']

### CODE REVIEW

**Comment by ocsig** [comment]
#173 This should patch the problem:
We left the indexes in the hash generation which made a new hash for every resource when the list got changed. Indexes are no longer part of the hash, so reordering or deletion works as expected.

File: dm/templates/iam_member/iam_member.py [hunk 1]
```diff
@@ -21,29 +21,26 @@ def generate_config(context):
 
     folder_id = context.properties.get('folderId')
     org_id = context.properties.get('organizationId')
-    project_id = context.properties.get('projectId')
-    if not folder_id and not org_id:
-        project_id = context.env['project']
+    project_id = context.properties.get('projectId', context.env['project'])
 
     resources = []
     for ii, role in  enumerate(context.properties['roles']):
```
Comment (sjvanrossum) [inline dm/templates/iam_member/iam_member.py]:
```suggestion
    for role in context.properties['roles']:
```

File: dm/templates/iam_member/iam_member.py [hunk 2]
```diff
@@ -21,29 +21,26 @@ def generate_config(context):
...
         for i, member in enumerate(role['members']):
-            suffix =  sha1('{}-{}-{}-{}'.format(role['role'], member, ii, i)).hexdigest()[:10]
+            suffix =  sha1('{}-{}'.format(role['role'], member)).hexdigest()[:10]
```
Comment (sjvanrossum) [inline dm/templates/iam_member/iam_member.py]:
```suggestion
            suffix = sha1('{}-{}'.format(role['role'], member)).hexdigest()[:10]
```

File: dm/templates/iam_member/iam_member.py [hunk 3]
```diff
@@ -21,29 +21,26 @@ def generate_config(context):
...
     folder_id = context.properties.get('folderId')
     org_id = context.properties.get('organizationId')
-    project_id = context.properties.get('projectId')
-    if not folder_id and not org_id:
-        project_id = context.env['project']
+    project_id = context.properties.get('projectId', context.env['project'])
 
     resources = []
     for ii, role in  enumerate(context.properties['roles']):
         for i, member in enumerate(role['members']):
```
Comment (sjvanrossum) [inline dm/templates/iam_member/iam_member.py]:
```suggestion
        for member in role['members']:
```

**Pull request merged**